File: backend/model/gif.py
from dataclasses import dataclass
from enum import Enum


class Rating(Enum):
    LEVEL_1 = 'g'
    LEVEL_2 = 'pg'
    LEVEL_3 = 'pg-13'
    LEVEL_4 = 'r'


@dataclass
class Gif:
    name: str = ''
    url: str = ''
    rating: str = ''
    creator: str = ''

    def __repr__(self) -> str:
        return f'{self.name} - {self.creator} - {self.url}'

    def to_dict(self) -> dict:
        return {
            'name': self.name,
            'url': self.url,
            'rating': self.rating,
            'creator': self.creator,
        }

File: frontend/src/app/model/gif.ts
export class Gif {
  public name: string = '';
  public creator: string = '';
  public url: string = '';
  public rating: string = '';
  public isFavourite: boolean = false;

  public toJson() {
    return {
      name: this.name,
      creator: this.creator,
      url: this.url,
      rating: this.rating,
    };
  }

  public static fromJson(json: any) {
    const gif = new Gif();
    gif.name = json.name;
    gif.creator = json.creator;
    gif.url = json.url;
    gif.rating = json.rating;
    return gif;
  }

  public toString() {
    return this.name;
  }
}
